### DISCUSSION TITLE
Regarding support for new versions of Bitcoin Core

### DISCUSSION THREAD

**Discussion by dscotese** [ACINQ/eclair#1961] [Node operators]
I received an email on Sept. 13th indicating that version 22.0 of Bitcoin core has been released.

It may be the case that Eclair strictly avoids using deprecated functions in Bitcoin Core, and that the Bitcoin Core team always supports deprecated functions for a major version or two.  In that case, Eclair would be expected to work with the "next version" or "next two versions" of Bitcoin Core, and a note detailing this could be added to https://github.com/ACINQ/eclair/blob/master/README.md.

On the other hand, either team may not be as strict as is necessary for that advice to be useful, and so a note detailing that Eclair is NOT forward compatible with new versions of Bitcoin Core could be added, and when the team has an idea of when a new Bitcoin core version will be supported, that ETA could be added to the README file too.

I think the most important thing to add to the README at this time is a note like this:
_Please do not upgrade your BItcoin Core software until Eclair supports the newer version._
To that end, I'll make a PR for it.

**Comment by t-bast** [comment]
Yes, it's completely impossible to claim that we would be compatible with *future* version of Bitcoin Core, as we simply cannot know what changes. The README already clearly says which version of Bitcoin Core are supported.

We will add support for the next version once the minor patch release is out (22.1) as we did with previous Bitcoin Core releases, and we will update the README accordingly.

**Comment by viaj3ro** [comment]
Is the current version of eclair compatible with core 22.0?

**Reply by t-bast** [reply]
We haven't tested it, so we can't guarantee it.
We will test and update eclair once core 22.1 is out.

**Reply by dscotese** [reply]
If you want, you can follow the instructions at https://github.com/ACINQ/eclair/blob/master/BUILD.md **without using the -DskipTests** and if it passes, then it should be okay, depending on how thorough the tests are.  There's always a chance that it won't be, but if it passes the tests and then fails somehow in production, then we need at least one new test to test that situation.

**Comment by evd0kim** [comment]
@viaj3ro it works. I do not know all implications.
